pypeclub/OpenPype
# Nuke: extract baked review videos presets

Client had requested two different review video looks - with baked viewer input process and without. To be able to do this we have to refactor the way nuke is baking mov file and enhance extract plugin with presets. Those allowing to set baking parameters per task-type and family filters. 

This PR includes:
- [x] Nuke Extract review baked mov with presets
- [x] Nuke ImageIO settigs had been enriched with  `project_anatomy/imageio/nuke/baking/viewerProcess` as the baking is usually needed to be in `rec709` rather then in sRGB 
![image](https://user-images.githubusercontent.com/40640033/141962818-892ad63a-b5d4-4487-8521-a0b40e64e077.png)

- [x] Global extract review settings added burnin profile linking
- [x] Global extract burnin accepting any linked profiles defined in representation data. 
- [x] Global extract burnin script had been fixed to be able to read source timecode from its metadata

# Known issues caused by this PR:
When representation is created the name is dynamically joined upstream from ExtractReviewDataMov down to burnins. But Load clip in Nuke is not able to recognize those representation names as it is working only with extract review profile names. 

# Testing:
1. first of all you had to open settings and define couple different profiles `project_settings/nuke/publish/ExtractReviewDataMov/outputs`. For example 
![image](https://user-images.githubusercontent.com/40640033/141808624-d52c266f-5b10-4936-8612-88146ee8bd47.png)
2. Create one more Burnin profile and change a little bit burnins so they are different then the default.  
3.  Create one more additional Extract review profile's output (could be copy of h264) but name it differently.  Ideally use following mxf settings 
![image](https://user-images.githubusercontent.com/40640033/141810164-e67d234f-2ba4-460e-9758-c880eca506c3.png)

4. In new output you had just created go to `Link to a burnin by name` and add the name of the burnin profile you had just crated in 2. 
![image](https://user-images.githubusercontent.com/40640033/141809623-17694664-11fd-4acc-9a66-eec390604b53.png)
  
5. Then open in nuke some of yours testing scenes
6. Create Render node and set it to plublish with Local rendering  and hit publish.
7. At the end you go to plublished Render folder and see all those review videos are there with the burnins you had set.
